### PR TITLE
jamtis: implement dense/sparse view privacy changes

### PR DESCRIPTION
**Pull request by jeffro256**
Description of changes can be found here: https://gist.github.com/tevador/50160d160d24cfc6c52ae02eb3d17024?permalink_comment_id=4665372#gistcomment-4665372

### CODE REVIEW

File: tests/performance_tests/view_scan.h [hunk 1]
```diff
@@ -350,19 +354,19 @@ class test_remote_scanner_client_scan_sp
...
-            sp::jamtis::address_index_t j_temp;
+            sp::jamtis::sparse_view_tag_t svt_temp;
             do
             {
-                sp::jamtis::gen_address_tag(m_basic_records.back().nominal_address_tag);
-            } while(sp::jamtis::try_decipher_address_index(*m_cipher_context,
-                m_basic_records.back().nominal_address_tag,
-                j_temp));
+                svt_temp = sp::jamtis::gen_sparse_view_tag();
+            } while(svt_temp != sp::sparse_view_tag(m_basic_records.back().enote));
```
Comment (UkoeHB) [inline tests/performance_tests/view_scan.h]: This does not look right, I think you broke this test.

**Review by UkoeHB — CHANGES_REQUESTED** [review]
Just skimmed it for now. I recommend running comparative perf tests for scanning before/after this PR and adding them to the PR comments.

File: tests/performance_tests/view_scan.h (hunk 1 above)
Comment (jeffro256) [reply tests/performance_tests/view_scan.h]: You're right about that...

**Comment by jeffro256** [comment]
Here's the results of `test_remote_scanner_client_scan_sp` performance test after fixing it:
```
seraphis_lib (original) run #1
test_remote_scanner_client_scan_sp (23 calls) - OK: 13434 µs/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 13217 µs/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 14304 µs/call

seraphis_lib (original) run #2
test_remote_scanner_client_scan_sp (23 calls) - OK: 13086 µs/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 13130 µs/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 14217 µs/call

jamtis_fix_fr_privacy (new PR) run #1
test_remote_scanner_client_scan_sp (23 calls) - OK: 1738 ms/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 1738 ms/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 1739 ms/call

jamtis_fix_fr_privacy (new PR) run #2
test_remote_scanner_client_scan_sp (23 calls) - OK: 1731 ms/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 1750 ms/call
test_remote_scanner_client_scan_sp (23 calls) - OK: 1734 ms/call
```

Ran on an AMD Ryzen 9 3900X 12-core CPU. So processing basic records from a remote scanner is ~128x slower (about what was predicted from Tevador's cycle estimates).

**Comment by jeffro256** [comment]
One performance difference that works in the new changes favor, though, is that basic records are 9% smaller because the nominal address index is not transmitted.

**Comment by UkoeHB** [comment]
If my math is right, the updated scanning for a scanner client is 1s per 10mill on-chain enotes (40k are scanned locally). 40k enotes transmitted for local scanning is at least 5-10MB so the overall cost is comparable with modern download speeds.

**Comment by jeffro256** [comment]
Closing in favor of #26

**Pull request closed without merging**